### ISSUE TITLE
Invalid bitcast  AWSServiceConfiguration LLVM ERROR: Broken function found, compilation aborted!

### ISSUE THREAD

**Issue by helloniklas** [aws/aws-sdk-ios#585]
So, recently I'm getting an error when Archiving for Release. It works fine to build the app for test debug on the device. But Archive comes up with this error.

Invalid bitcast
  %45 = bitcast %CSo23AWSServiceConfiguration* %43 to i64, !dbg !1163
LLVM ERROR: Broken function found, compilation aborted!

Any ideas what's going on?

**Comment by helloniklas** [comment]
Setting the Optimization Level to None[-Onone] for Release goes around this problem.

**Comment by minbi** [comment]
Thanks for reporting this issue and providing a workaround.

**Comment by sclark39** [comment]
I am getting this error too while trying to use the APIGatewayClientBridge.h file. The generated swift code was not updated to Swift 3 at all, but after getting rid of all the compile issues this is what I am left with.

Setting Optimization level to none is not fixing it.

**Comment by helloniklas** [comment]
@sclark39 my problem was on the same line I believe. However, setting on the build schemes to use none for optimisation got around it for me, did you try and remove all deprived build files etc from Xcode library folder? Although setting to none is not a solution. AWS team need to sort out Swift compatibility as a priority if you ask me.

**Comment by sclark39** [comment]
@helloniklas I did run a clean and then attempted building again but maybe it left something behind. I'll see if there's anything else I can clean up and try another build today.

Would you be able to share your version of the client which was causing the issue? I assume you also had to rewrite it for Swift 3, and if so maybe you did it a little differently than me which the compiler was happier about.

I agree. At this point which AWS' Swift support where it's at I fear my only option may be to implement the signing v4 part myself and totally bypass the APIGateway.framework.

**Comment by sclark39** [comment]
Confirmed even with a clean/build and double checking all the derived files were deleted I can't build in the Debug nor Release configurations with optimization set to none..

It looks like my two other options, if I can't work around this compiler error, would be:
1. Figure out how to work with AWSSignatureSignerUtility directly
2. Build a functional APIGatewayClient in Obj-C to reduce the need to send data across the obj-c / swift barrier

**Comment by sclark39** [comment]
I tried compiling this with the development version of swift from https://swift.org/

Doing so got me a much more useful error:

`error: cannot assign to property: 'configuration' is a get-only property`

I believe the issue is that in the AWSAPIGateway.framework it is marked as read only, and that Swift doesn't allow changing the access in a bridging header, as of Swift 3.0.

I have no way to test this myself yet. I will need to get set up to be build the AWS framework from source to do so.

**Comment by sclark39** [comment]
**Workaround:** 
1. Use the Obj-C generated API Gateway Client SDK instead of the Swift ones. 
1. Include your API Gateway header file in the bridging header
1. You now have a fully compatible swift 3 api gateway client

**Comment by helloniklas** [comment]
@sclark39 that still does not work for me. I'm not even using any API Gateway generated classes but need to include a bridging AWSAPIGatewayClient.h to be able to use invokeHTTPRequest from Swift which is included.

**Comment by sclark39** [comment]
I generated the Obj-C API, and then modified it to add access to invokeHTTPRequest via the new class... I also rewrote invokeHTTPRequest to accept the body as a dictionary instead of a model, and to return the json dictionary rather than try to parse into an AWSModel.

.m file
https://gist.github.com/sclark39/6b03820eba8a26ace6bb97fbccd121a9

.h file
https://gist.github.com/sclark39/e719efcd40449fc174fee7e1b3be1f25

I just included the header file in the bridging header, and then was able to do the following from swift:

https://gist.github.com/sclark39/4da054d0519612dcb55408a2f8b1f9ba

**Comment by sclark39** [comment]
What I found was that creating a bridging header that changed configuration to be writable when it is already defined as read-only was causing a compiler issue with Swift. Doing so in a subclass of the AWSAPIGatewayClient works fine.

Another approach to fixing this I believe would be to just change the AWSAPIGatewayClient source directly so that it  is declared as writable in the header from the start, but that requires you to be set up to build the AWS SDK from source.

**Comment by karthiksaligrama** [comment]
thanks will be fixed in the next version of the sdk.

Also if you want to generically invoke any api you could use the generic invoker method thats in the sdk.

```
 let bodyString = "{\"key3\": \"value3\",\"key2\": \"value2\",\"key1\": \"value1\"}"
 let stringBodyRequest = AWSAPIGatewayRequest(httpMethod: "POST",
                                                     urlString: "/TestFunction",
                                                     queryParameters: nil,
                                                     headerParameters: headerParameters,
                                                     httpBody: bodyString)
        
client!.invoke(stringBodyRequest)
```

More details in [this test case](https://github.com/aws/aws-sdk-ios/blob/master/AWSAPIGatewayTests/AWSAPIGatewayInvokeTest.swift)

**Comment by sclark39** [comment]
Oh, well that would have been a whole lot simpler!

**Comment by helloniklas** [comment]
@karthiksaligrama although I am using that method, and still get Swift compiler errors as outlined above, unless I switch off optimisation. Hopefully the Swift implementation will work better i the next release.

**Comment by karthiksaligrama** [comment]
resolved with 2.5.3
